### PR TITLE
Add more test of  DatabaseProtocolFrontendEngineFactory

### LINKED ISSUE #7012: Add more test cases for DatabaseProtocolFrontendEngineFactory when newInstance success

### PR DESCRIPTION
**Pull request by xbkaishui**
Fixes #7012.

Changes proposed in this pull request:
- Add test case when newInstance is not null

### CODE REVIEW

**Review by terrymanu — CHANGES_REQUESTED** [review]
Please fix CI

**Comment by xbkaishui** [comment]
done

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java [hunk 1]
```diff
@@ -18,12 +18,35 @@
...
+import org.apache.shardingsphere.proxy.frontend.spi.DatabaseProtocolFrontendEngine;
+import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
 
+import java.lang.reflect.Method;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java:32]: Why use `@RunWith(MockitoJUnitRunner.class)` if not filed use @Mock?

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java [hunk 2]
```diff
@@ -18,12 +18,35 @@
...
+        method.invoke(null, DatabaseProtocolFrontendEngine.class, new MockDatabaseProtocolFrontendEngine());
+    }
+    
     @Test(expected = UnsupportedOperationException.class)
     public void assertNewInstanceWhenUnsupported() {
         DatabaseProtocolFrontendEngineFactory.newInstance(DatabaseTypes.getActualDatabaseType("Oracle"));
     }
+    
+    @Test
+    public void assertNewInstanceMysql() {
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java:48]: It is better to rename `Mysql` to `MySQL`

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java [hunk 3]
```diff
@@ -18,12 +18,35 @@
...
+        DatabaseProtocolFrontendEngine databaseProtocolFrontendEngine = DatabaseProtocolFrontendEngineFactory.newInstance(new MySQLDatabaseType());
+        Assert.assertNotNull(databaseProtocolFrontendEngine);
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java:50]: Please use static import for `Assert.assertNotNull`

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/MockDatabaseProtocolFrontendEngine.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+import org.apache.shardingsphere.db.protocol.codec.DatabasePacketCodecEngine;
+import org.apache.shardingsphere.infra.database.type.dialect.MySQLDatabaseType;
+import org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.BackendConnection;
+import org.apache.shardingsphere.proxy.frontend.context.FrontendContext;
+import org.apache.shardingsphere.proxy.frontend.engine.AuthenticationEngine;
+import org.apache.shardingsphere.proxy.frontend.engine.CommandExecuteEngine;
+import org.apache.shardingsphere.proxy.frontend.spi.DatabaseProtocolFrontendEngine;
+
+public class MockDatabaseProtocolFrontendEngine implements DatabaseProtocolFrontendEngine {
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/MockDatabaseProtocolFrontendEngine.java:28]: Do we really need this class? Can we use mock to instead this one?

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/MockDatabaseProtocolFrontendEngine.java (hunk 4 above)
Comment (xbkaishui) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/MockDatabaseProtocolFrontendEngine.java:28]: Nop, Currently implement of DatabaseProtocolFrontendEngineFactory is using class newInstance method to renew a instance of class, it is hard to mock.

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java (hunk 2 above)
Comment (xbkaishui) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java:48]: done

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java (hunk 1 above)
Comment (xbkaishui) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java:32]: sorry , forgot to remove

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java (hunk 3 above)
Comment (xbkaishui) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java]: done

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/MockDatabaseProtocolFrontendEngine.java (hunk 4 above)
Comment (xbkaishui) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/MockDatabaseProtocolFrontendEngine.java:28]: I went deep into the mockito fr
amework,  it is not easy to mock static method, 
If we want to use this, we need update version to 3.x,  is it necessary?

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/MockDatabaseProtocolFrontendEngine.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    
+    public MockDatabaseProtocolFrontendEngine() {
+    }
+    
+    @Override
+    public String getDatabaseType() {
+        return new MySQLDatabaseType().getName();
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/MockDatabaseProtocolFrontendEngine.java:35]: It is better to define a new type here, `FIXTURE` maybe a good name than `MySQL`

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/MockDatabaseProtocolFrontendEngine.java (hunk 4 above)
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/MockDatabaseProtocolFrontendEngine.java:28]: The class `MockDatabaseProtocolFrontendEngine` is not really mocked. `DatabaseProtocolFrontendEngineFixture` may better.

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java [hunk 6]
```diff
@@ -18,12 +18,22 @@
...
+
 public final class DatabaseProtocolFrontendEngineFactoryTest {
     
     @Test(expected = UnsupportedOperationException.class)
     public void assertNewInstanceWhenUnsupported() {
         DatabaseProtocolFrontendEngineFactory.newInstance(DatabaseTypes.getActualDatabaseType("Oracle"));
     }
+    
+    @Test
+    public void assertNewInstanceMySQL() {
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/test/java/org/apache/shardingsphere/proxy/frontend/DatabaseProtocolFrontendEngineFactoryTest.java:35]: The name of `assertNewInstanceMySQL` is inaccurate, please consider change it.

**Review by terrymanu — CHANGES_REQUESTED** [review]

**Review by terrymanu — APPROVED** [review]

**Review by terrymanu — APPROVED** [review]
